### DISCUSSION TITLE
[Feature Request]:

### DISCUSSION THREAD

**Discussion by developerabdan** [beyondcode/herd-community#190] [Q&A]
### Feature Description

How to work with subdomain in laravel?

**Comment by tsangaris** [comment]
I would like to know this as well. Currently I used this "composer create-project laravel/laravel api.domain" and to navigate to the site i use https://api.domain.test. But I don't know if this the correct approach and if it infact acts as a subdomain.

**Reply by meduzen** [reply]
In case you have a project having both a domain and subdomains:

As explained in [Laravel routing docs](https://laravel.com/docs/10.x/routing#route-group-subdomain-routing), first make sure to encapsulate your routes declaration by `Route::domain`:

```php
<?php // routes/web.php

Route::domain('your-subdomain.domain.test')
    ->get('/', 'SomeController')->name('someRouteName');
}
```

In earlier versions of Laravel Valet, you’d needed to create a symlink using `ln -s example your-subdomain.example` in the parent folder of your project folder. **Now it’s not needed anymore**, so you don’t need anything else than `Route::domain`, whether you are using Valet or Herd.

Note: there are strategies to support a `.test` domain, `localhost` or your online domain as well, so local development is not a struggle (you don’t want to hardcode a `.test` domain in the route declarations). I’ll write about it this week-end (hopefully) and post the link here.

**Reply by pfried** [reply]
Is there any way to work with subdomains? I currently need to to `herd link subdomain.domain.test` for it to work (Windows 10). It works if I do that manually, but that is cumbersome

**Reply by mpociot** [reply]
Windows does not support wildcard subdomains due to the fact that we don't use something like DNSMasq but instead modify your hosts file. So you could either manually link the site, or create the hosts entries yourself.

**Comment by MPGs007** [comment]
Seen several questions querying the same thing, on windows we have use of acrylic DNS which enables / allows us to use wildcard domains for localhost  - 127.0.0.1 *.domain.test  however this is not available in herd - each subdomain has to be added manually to the hosts file - for testing / creating a multi-tenant application with multiple databases accessed via subdomains this isn't a great solution - does anyone have knowledge how to get Herd working with acrylic DNS server? please

**Comment by MPGs007** [comment]
Hopefully this will help - diagnosed the issue using acrylic DNS - the wildcard *.domain.test was not hitting the acrylic DNS server my windows wifi adapter was sending it via the router first which caused failure. 
Changed my wifi network adapter for IPv4 to use 127.0.0.1 as primary DNS and set secondary as google 8.8.8.8 and now acrylic is able to handle wildcards
